docs-source/src/architecture.rst

```diff
@@ -29,34 +29,35 @@ The ``license-manager-agent`` is also responsible for making booking requests to
 when Slurm is configured to use the ``PrologSlurmctld`` script provided by ``license-manager-agent``.
 
 Each job submitted to Slurm will make a request to the ``license-manager-backend`` to book the needed licenses prior
-to the allocation of the job. The booking ensures that the licenses are available for the job to use, taking into
+to the allocation of the job. The booking ensures that the licenses are available for the job to use by taking into
 consideration the licenses booked for other jobs and the license usage in the license server.
 
 If the booking cannot be made, the job will be kept in the queue until there are enough licenses available to
 satisfy the booking request.
 
 Grace time
 **********
-Since a job can take some time to check out the license from the license server, each license has a ``grace time``
-period used to indicate when the booking can be deleted, since the job will have already checked out the license
-from the license server and doesn't need to hold the licenses in the cluster anymore.
+A job can take some time to check out the license from the license server after it is submitted to Slurm.
+Thrus, each license has a ``grace time`` period that is used to indicate how long a booking will be retained.
+After the ``grace time`` expires, the booking is deleted. This means that the license was checked out from the
+license server and doesn't need a booking anymore.
 
-The booking is deleted once it reaches the ``grace time`` period for the license or when the job finishes, whichever comes first.
+A booking is deleted after the job finishes even if there is remaining ``grace time``.
```

Comment:
I think this sentence isn't needed any more after the previous change.

Reply:
100%, I removed it.